export.go

```diff
@@ -59,18 +106,34 @@ func ExportToRipoff(ctx context.Context, tx pgx.Tx, excludeTables []string) (Rip
 	missingDependencies := []RowMissingDependency{}
 
 	for table, primaryKeys := range primaryKeyResult {
-		columns := make([]string, len(foreignKeyResult[table].Columns))
-		// Due to yaml limitations, ripoff treats all data as nullable text on import and export.
-		for i, column := range foreignKeyResult[table].Columns {
-			columns[i] = fmt.Sprintf("CAST(%s AS TEXT)", pq.QuoteIdentifier(column))
+		// Filter out excluded columns from the foreign key result columns
+		var filteredColumns []string
+		var filteredColumnNames []string
+		for _, column := range foreignKeyResult[table].Columns {
+			if !shouldExcludeColumn(table, column, tableSpecificExclusions, globalColumnExclusions) {
+				filteredColumns = append(filteredColumns, fmt.Sprintf("CAST(%s AS TEXT)", pq.QuoteIdentifier(column)))
+				filteredColumnNames = append(filteredColumnNames, column)
+			}
 		}
-		selectQuery := fmt.Sprintf("SELECT %s FROM %s;", strings.Join(columns, ", "), pq.QuoteIdentifier(table))
+
+		// Skip table if no columns remain after filtering
+		if len(filteredColumns) == 0 {
+			continue
+		}
+
+		selectQuery := fmt.Sprintf("SELECT %s FROM %s;", strings.Join(filteredColumns, ", "), pq.QuoteIdentifier(table))
 		rows, err := tx.Query(ctx, selectQuery)
 		if err != nil {
 			return RipoffFile{}, err
 		}
 		defer rows.Close()
 		fields := rows.FieldDescriptions()
+
+		// Verify that field descriptions match our filtered columns (sanity check)
+		if len(fields) != len(filteredColumnNames) {
+			return RipoffFile{}, fmt.Errorf("mismatch between query fields (%d) and filtered columns (%d) for table %s", len(fields), len(filteredColumnNames), table)
+		}
+
```

Comment:
I don't totally understand when this wouldn't be true, or otherwise why `filteredColumnNames` exists

Reply:
Oh yeah this should be pruned now. I don't think it's possible to end up with the mismatch here. I had it in place originally before I added the tests and forgot to pull it out.